codegenerator/cli/npm/envio/src/Platform.res

```diff
@@ -0,0 +1,133 @@
+type name = | @as("evm") Evm | @as("fuel") Fuel
+
+type t = {
+  name: name,
+  blockFields: array<string>,
+  transactionFields: array<string>,
+  blockNumberName: string,
+  blockTimestampName: string,
+  blockHashName: string,
+  getNumber: Internal.eventBlock => int,
+  getTimestamp: Internal.eventBlock => int,
+  getId: Internal.eventBlock => string,
+  cleanUpRawEventFieldsInPlace: Js.Json.t => unit,
+}
+
+module Evm = {
+  @get external getNumber: Internal.eventBlock => int = "number"
+  @get external getTimestamp: Internal.eventBlock => int = "timestamp"
+  @get external getId: Internal.eventBlock => string = "hash"
+  
+  let cleanUpRawEventFieldsInPlace: Js.Json.t => unit = %raw(`fields => {
+    delete fields.hash
+    delete fields.number
+    delete fields.timestamp
+  }`)
+}
+
+let evm: t = {
+  name: Evm,
+  blockFields: [
+    "number",
+    "timestamp",
+    "hash",
```

Comment:
Haven't found where this is used yet but would love it if this would come from the client.